gradle-baseline-java/src/test/groovy/com/palantir/baseline/BaselineFormatIntegrationTest.groovy

```diff
@@ -221,4 +221,41 @@ class BaselineFormatIntegrationTest extends AbstractPluginTest {
         BuildResult result = with('spotlessJavaCheck').build()
         result.task(":spotlessJava").outcome == TaskOutcome.SUCCESS
     }
+
+    def 'formatDiff updates only lines changed in git diff'() {
+        when:
+        buildFile << standardBuildFile
+        "git init".execute(Collections.emptyList(), projectDir).waitFor()
+        "git config user.name Foo".execute(Collections.emptyList(), projectDir).waitFor()
+        "git config user.email foo@bar.com".execute(Collections.emptyList(), projectDir).waitFor()
+
+        file('src/main/java/Main.java') << '''
+        class Main {
+            public static void crazyExistingFormatting  (  String... args) {
+
+            }
+        }
+        '''.stripIndent()
+
+        "git add .".execute(Collections.emptyList(), projectDir).waitFor()
+        "git commit -m Commit".execute(Collections.emptyList(), projectDir).waitFor()
+
+        file('src/main/java/Main.java').text = '''
+        class Main {
+            public static void crazyExistingFormatting  (  String... args) {
+                                        System.out.println("Reformat me please");
+            }
+        }
```

Comment:
I would make this example super minimal just so if the formatter changes any more it won't affect this test